### PR TITLE
Making it work on macOS  #112

### PR DESCRIPTION
**Pull request by romanr**

### CODE REVIEW

File: requirements-torch.txt [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-torchvision; sys_platform == "linux"
+--index-url https://download.pytorch.org/whl/cpu ; sys_platform == "linux"
+torch==2.3.1; sys_platform == "linux"
+torchvision==0.18.1; sys_platform == "linux"
 
 # Windows
---index-url https://download.pytorch.org/whl/cu118 ; sys_platform == "win32"
+--index-url https://download.pytorch.org/whl/cpu ; sys_platform == "win32"
 torch==2.3.1; sys_platform == "win32"
-torchvision==0.18.1; sys_platform == "win32"
```
Comment (nagadomi) [inline requirements-torch.txt:13]: I think no need to change requirements-torch.txt

This change makes CUDA (GPU) unavailable on Linux and Windows.
According to https://pytorch.org/get-started/locally/ , --index-url is not required for macOS.

I have tested that the current requiements-torch.txt works fine on Linux and Windows CPU/GPU environments.
Also I have tested that it works on macOS, but that was over a year ago.

**Comment by nagadomi** [comment]
Have you actually tested that it works on your macOS+MPS environment?

I have been careful not to add code that does not work on macOS since https://github.com/nagadomi/nunif/issues/112, but I have not tested it on macOS for the last few years.

**Comment by romanr** [comment]
Yes, tested, and that's why I am submitting :)
image

**Comment by romanr** [comment]
> If this works without changes for requirements-torch.txt, I will merge this PR.
> 
> Also, StereoFormat=VR90 perhaps does not work and there are a few ideas to prevent fallback to CPU as much as possible. So I am planning to test on macOS soon on my end.

Yes this PR works on macOS, with this sequence of commands:
```sh
python3 -m venv .venv
source .venv/bin/activate
pip3 install -r requirements-torch.txt
pip3 install -r requirements.txt
pip3 install -r requirements-gui.txt
python -m iw3.gui
```

I wanted to add that to README but not sure which one. Install-Mac.md was not linked from main document, so I found it only after it was done. 
Install instructions are usually in one file for all platforms. If you want, I can organise all into one file and have separate development instructions as it usually goes.

File: requirements-torch.txt (hunk 1 above)
Comment (nagadomi) [reply requirements-torch.txt]: It seems that the platform condition cannot be used for --index-url option.
This is my mistake and I will fix it in some way on my end.

File: requirements-torch.txt
Comment (romanr) [inline requirements-torch.txt:1]: I pushed change the solution that I found to use --extra-index-url instead

**Comment by nagadomi** [comment]
I merged this to master with cherry-pick.
https://github.com/nagadomi/nunif/commit/584e55c3bf11a1e61bf9cceb12b8632e361d3bf2

Additional changes
https://github.com/nagadomi/nunif/commit/87b3089c46a0e314f1f8a00ad5be5eabe84ba575

One problem that cannot be fixed is segmentation fault when exiting GUI.
It is a bug in wxpython and already fixed, but pip is not updated.
https://github.com/wxWidgets/Phoenix/issues/2455

**Comment by nagadomi** [comment]
Currently FP16 option is ignored (It is always FP32).
When MPS autocast is supported on PyTorch, it can be supported on our side as well.
https://github.com/pytorch/pytorch/pull/99272

**Pull request closed without merging**